src/filesystem-upload-options.js

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import DirectBinaryUploadOptions from './direct-binary-upload-options';
+import { DefaultValues } from './constants';
+
+/**
+ * Options specific to a file system upload. Also supports all options defined by
+ * DirectBinaryUploadOptions.
+ */
+export default class FileSystemUploadOptions extends DirectBinaryUploadOptions {
+
+    /**
+     * Creates a new FileSystemUploadOptions instance that will have the same options
+     * as an existing options instance.
+     * @param {DirectBinaryUploadOptions} uploadOptions Options whose value should
+     *  be copied.
+     */
+    static fromOptions(uploadOptions) {
+        const newOptions = new FileSystemUploadOptions();
+        newOptions.options = { ...uploadOptions.options };
+        newOptions.controller = uploadOptions.controller;
+        return newOptions;
+    }
+
+    /**
+     * Sets the maximum number of files that can be uploaded using the module at one time. If
+     * the total number of files exceeds this number then the library will throw an exception
+     * with code TOO_LARGE.
+     * @param {*} maxFileCount 
+     * @returns {FileSystemUploadOptions} The current options instance. Allows for chaining.
+     */
+    withMaxUploadFiles(maxFileCount) {
+        this.options.maxUploadFiles = maxFileCount;
+        return this;
```

Comment:
Why return `this`, since this is an instance method (not static)?

Reply:
It's for convenience, and is how `DirectBinaryUploadOptions` already works. It allows chaining:

```
options.withMaxUploadFiles(10)
  .withDeepUpload(true)
  .withUrl(someUrl);
```

There is a brief description of the return value in the method's documentation.

Reply:
Aaaaaah, it's for chaining. I like the design, then :)